crates/debug/types/src/source_manager.rs

```diff
@@ -254,16 +254,27 @@ pub trait SourceManagerExt: SourceManager {
 #[cfg(feature = "std")]
 impl<T: ?Sized + SourceManager> SourceManagerExt for T {}
 
+// TODO rename to reflect the fact it's a "glue" trait that adapts per target, but only after
+// the `-base` has been validated to work with this approach.
 /// [SourceManagerSync] is a marker trait for [SourceManager] implementations that are also Send +
 /// Sync, and is automatically implemented for any [SourceManager] that meets those requirements.
 ///
 /// [SourceManager] is a supertrait of [SourceManagerSync], so you may use instances of the
 /// [SourceManagerSync] where the [SourceManager] is required, either implicitly or via explicit
 /// downcasting, e.g. `Arc<dyn SourceManagerSync> as Arc<dyn SourceManager>`.
+
+#[cfg(target_arch = "wasm32")]
 pub trait SourceManagerSync: SourceManager + Send + Sync {}
 
+#[cfg(target_arch = "wasm32")]
 impl<T: ?Sized + SourceManager + Send + Sync> SourceManagerSync for T {}
 
+#[cfg(not(target_arch = "wasm32"))]
+pub trait SourceManagerSync: SourceManager + Send {}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl<T: ?Sized + SourceManager + Send> SourceManagerSync for T {}
```

Comment:
Do we still need these changes in this PR?

Reply:
Removed